### PR TITLE
feat: progress and footer stepper

### LINKED ISSUE #20: FooterStepper: Create footer-stepper component
[Figma Design](https://www.figma.com/file/msxmKajRPLYiDDmsP60Ndh/Design-System-v1.1?type=design&node-id=6576-51238&t=ttVz3vzOxnYJ4iYn-4)

should support:

- [x] different formats based on prop

### LINKED ISSUE #44: Progress: Create progress component
[Figma Design](https://www.figma.com/file/msxmKajRPLYiDDmsP60Ndh/Design-System-v1.1?type=design&node-id=597-47925&mode=dev)

should support:

- [x] buffer type
- [x] determinate type
- [x] indeterminate type
- [x] primary / secondary / inherit color modes
- [x] horizontal progress
- [x] circular progress
- [x] light / dark mode
- [x] progress label

### PR DESCRIPTION
**Pull request by tewshi**
Closes #20 
Closes #44

### CODE REVIEW

**Review by lukicenturi — APPROVED** [review]
LGTM!

File: src/components/steppers/FooterStepper.vue [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      <rui-button
+        :disabled="modelValue <= 1"
+        color="primary"
+        tile
+        variant="text"
+        @click="onPrev()"
+      >
+        <template #prepend>
+          <rui-icon class="-ml-[.8rem]" name="arrow-left-s-line" />
+        </template>
```
Comment (lukicenturi) [inline src/components/steppers/FooterStepper.vue:69]:
```
        <template #prepend>
          <rui-icon size="18" name="arrow-left-s-line" />
        </template>
```

**Review by lukicenturi — CHANGES_REQUESTED** [review]
Need little adjustment here
image

image


also the text here
image

and the white bullet here
image

File: src/components/steppers/FooterStepper.vue (hunk 1 above)
Comment (tewshi) [reply src/components/steppers/FooterStepper.vue:69]: let's reduce the size right, and remove negative margin? ok

File: src/components/buttons/Button.vue [hunk 2]
```diff
@@ -142,6 +142,8 @@ const usedElevation: ComputedRef<number | string> = computed(() => {
 
   &.text {
     @apply px-2;
+    @apply bg-transparent shadow-none;
+    @apply disabled:bg-transparent disabled:active:bg-transparent #{!important};
```
Comment (lukicenturi) [inline src/components/buttons/Button.vue:146]: at the line `135` I think I already put this line. Isn't it working?

File: src/components/buttons/Button.vue [hunk 3]
```diff
@@ -180,6 +182,10 @@ const usedElevation: ComputedRef<number | string> = computed(() => {
     &.sm {
       @apply px-2 py-2;
     }
+
+    &.text {
+      @apply disabled:bg-transparent disabled:active:bg-transparent #{!important};
```
Comment (lukicenturi) [inline src/components/buttons/Button.vue:187]: `fab` and `text` is unified in variant now. So they can have both class. Then this line will not be used at all.

Comment (tewshi) [reply src/components/buttons/Button.vue:187]: okay, on it, will also rename `type` to variant in progress and stepper

File: src/components/buttons/Button.vue [hunk 4]
```diff
@@ -141,7 +141,7 @@ const usedElevation: ComputedRef<number | string> = computed(() => {
   }
 
   &.text {
-    @apply px-2;
+    @apply px-2 shadow-none;
```
Comment (lukicenturi) [inline src/components/buttons/Button.vue:144]: is there any case where without `shadow-none`, it can have shadow?

**Review by lukicenturi — APPROVED** [review]
LGTM!

**Review by kelsos — APPROVED** [review]
lgtm

**Pull request merged**